lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -4,163 +4,507 @@
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderColumnHeader({
+            chevron: true,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          })!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+    });
+
+    describe('when the rowBtnIcon is set', function() {
+      const renderTableCell = (extraProps?: NxTableCellProps, isHeader?: boolean) => {
+        const children = (
+          <NxTable.Row isClickable={true} clickAccessibleLabel="dolphin">
+            <NxTable.Cell>Foo</NxTable.Cell>
+            <NxTable.Cell { ...extraProps } />
+          </NxTable.Row>
+        );
+
+        return render(
+          <NxTable>
+            { isHeader ? <NxTable.Head>{children}</NxTable.Head>
+            : <NxTable.Body>{children}</NxTable.Body>
+            }
+          </NxTable>
+        ).getAllByRole(isHeader ? 'columnheader' : 'cell')[1] as HTMLElement;
+      };
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderTableCell({
+            rowBtnIcon: faEdit,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          }, true)!;
 
-    await act(async () => {
-      component.setProps({
-        children: (
-          <>
-            <NxTableHead>
-              <NxTableRow>
-                <NxTableCell>Foo</NxTableCell>
-              </NxTableRow>
-            </NxTableHead>
-            <NxTableBody error="Error!"/>
-          </>
-        )
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+
+      describe('when not isHeader', function() {
+        it('does not show button when rowBtnIcon is undefined', function() {
+          expect(within(renderTableCell({ rowBtnIcon: undefined })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('does not show button when rowBtnIcon is null', function() {
+          expect(within(renderTableCell({ rowBtnIcon: null })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('show button when rowBtnIcon faEdit is set', function() {
+          expect(within(renderTableCell({ rowBtnIcon: faEdit })).getByRole('button')).toBeInTheDocument();
+        });
+
+        it('ignores the children and sort settings and adds the icon child wrapped in a nx-cell__row-btn',
+            function() {
+              const cell = renderTableCell({
+                rowBtnIcon: faEdit,
+                sortDir: 'asc',
+                children: <span>foo</span>
+              });
+
+              expect(within(cell).getByRole('button')).toBeInTheDocument();
+              expect(cell.innerHTML).not.toEqual('<span>foo</span>');
+            }
+        );
+
+        it('sets the nx-cell__row-btn aria-label from the RowContext', function() {
```

Comment:
Externally, it's not from the `RowContext` it's from the `clickAccessibleName` or from the row text content (both should be tested)